File: app/src/main/java/net/radityalabs/contactapp/presentation/util/PhoneUtil.java
package net.radityalabs.contactapp.presentation.util;

import android.content.Intent;
import android.net.Uri;

/**
 * Created by radityagumay on 4/16/17.
 */

public class PhoneUtil {

    public static Intent sms(String phoneNumber) {
        Intent intent = new Intent(android.content.Intent.ACTION_VIEW);
        intent.setType("vnd.android-dir/mms-sms");
        intent.putExtra("address", phoneNumber.trim());
        return intent;
    }

    public static Intent call(String phoneNumber) {
        String uri = "tel:" + phoneNumber.trim();
        Intent intent = new Intent(Intent.ACTION_CALL);
        intent.setData(Uri.parse(uri));
        return intent;
    }

    public static Intent email(String address) {
        Intent email = new Intent(Intent.ACTION_SENDTO);
        email.putExtra(Intent.EXTRA_EMAIL, new String[]{address});
        email.putExtra(Intent.EXTRA_SUBJECT, "subject");
        email.putExtra(Intent.EXTRA_TEXT, "message");
        email.setType("message/rfc822");
        return email;
    }
}

File: app/src/main/java/net/radityalabs/contactapp/presentation/util/PhoneUtil.kt
package net.radityalabs.contactapp.presentation.util

import android.content.Intent
import android.net.Uri

/**
 * Created by radityagumay on 4/16/17.
 */

object PhoneUtil {

    fun sms(phoneNumber: String): Intent {
        val intent = Intent(android.content.Intent.ACTION_VIEW)
        intent.type = "vnd.android-dir/mms-sms"
        intent.putExtra("address", phoneNumber.trim { it <= ' ' })
        return intent
    }

    fun call(phoneNumber: String): Intent {
        val uri = "tel:" + phoneNumber.trim { it <= ' ' }
        val intent = Intent(Intent.ACTION_CALL)
        intent.data = Uri.parse(uri)
        return intent
    }

    fun email(address: String): Intent {
        val email = Intent(Intent.ACTION_SENDTO)
        email.putExtra(Intent.EXTRA_EMAIL, arrayOf(address))
        email.putExtra(Intent.EXTRA_SUBJECT, "subject")
        email.putExtra(Intent.EXTRA_TEXT, "message")
        email.type = "message/rfc822"
        return email
    }
}
